Core/src/org/sleuthkit/autopsy/contentviewers/WindowsRegistryViewer.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Autopsy
+ *
+ * Copyright 2019 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Copyright 2013-2018 Willi Ballenthin
+ * Contact: willi.ballenthin <at> gmail <dot> com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.contentviewers;
+
+import com.williballenthin.rejistry.RegistryHive;
+import com.williballenthin.rejistry.RegistryHiveBuffer;
+import com.williballenthin.rejistry.RegistryParseException;
+import java.awt.BorderLayout;
+import java.awt.Component;
+import java.awt.Cursor;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.List;
+import java.util.logging.Level;
+import javax.swing.JPanel;
+import org.sleuthkit.autopsy.rejview.RejView;
+import org.sleuthkit.autopsy.coreutils.Logger;
+import org.sleuthkit.datamodel.AbstractFile;
+import org.sleuthkit.datamodel.Content;
+import org.sleuthkit.datamodel.TskCoreException;
+
+/**
+ * FileTypeViewer which displays information for Windows Registry files
+ */
+class WindowsRegistryViewer extends JPanel implements FileTypeViewer {
+
+    private static final long serialVersionUID = 1L;
+    private static final Logger logger = Logger.getLogger(WindowsRegistryViewer.class.getName());
+    private static final String[] SUPPORTED_MIMETYPES = new String[]{"application/x.windows-registry"};
+    //Registry log files which should be ignored share the same signature as Registry files but appear to have a size of 1024
+    private static final String LOG_FILE_EXTENSION = "log"; //base extension for log files
+    private RejView regview;
+    private AbstractFile lastFile;
+
+    WindowsRegistryViewer() {
+        super(new BorderLayout());
+    }
+
+    private void setDataView(Content content) {
+        if (content == null) {
+            this.resetComponent();
+            return;
+        }
+
+        this.setCursor(Cursor.getPredefinedCursor(Cursor.WAIT_CURSOR));
+
+        if (content.getSize() == 0) {
+            return;
+        }
+
+        byte[] data = new byte[(int) content.getSize()];
+
+        try {
+            content.read(data, 0x0, content.getSize());
+        } catch (TskCoreException ex) {
+            logger.log(Level.WARNING, "Failed to read file content.", ex);
```

Comment:
Shouldn't we exit here?

Reply:
Returning there sounds like a reasonable thing to do if the reading of the data fails. It wasn't being done in the original code I was integrating, but that is no indication it shouldn't be done. I will make this change and test to ensure it works, unless I hear otherwise.

Reply:
I don't really know what the right answer is so you should probably seek the opinion of @rcordovano.